### ISSUE TITLE
Product custom option delete while edit product programmatically

### ISSUE THREAD

**Issue by koushikch7** [magento/magento2#29572]
### Preconditions (*)

1. Magento 2.3.0
2. php 7.2.3

### Steps to reproduce (*)

1. Add custom options to any product. 
2. Update product SKU programmatically (custom script or custom module) that have custom options, for instance using following code:
```php
$productCollectionFactory = $objectManager->get('\Magento\Catalog\Model\ResourceModel\Product\CollectionFactory');
$storeId = 10;
$productCollection = $productCollectionFactory->create()->addAttributeToSelect('*')->addStoreFilter($storeId);
foreach ($productCollection as $product) {
    $sku = $product->getSku();
    $product->setSku('UpdatedSku-'.$sku);
    $product->save();
}
```
3.  check the same product in the backend

### Expected result (*)

1. The custom options should be unchanged/undeleted

### Actual result (*)

1. Custom option deletes / no more available
---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without the workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look, and feel, “quality” or “usability”._

**Comment by ihor-sviziev** [comment]
Seems like this is the same issue as described in https://github.com/magento/magento2/pull/29349

**Comment by koushikch7** [comment]
Hello @ihor-sviziev , I think #29349  is not related, because my problem is custom options are completely deleted/unlinked from the product store and default view. where has in the mentioned ticket #29349  the custom options are re-defining w.r.t store level. so still does this related?

**Comment by ihor-sviziev** [comment]
@koushikch7 could you provide sample code for reproducing this issue?

I believe it was fixed by removing those two lines: https://github.com/magento/magento2/pull/29349/files?file-filters%5B%5D=.js&file-filters%5B%5D=.xml#diff-d31bcdab992b8a290ce2d9ae7ed5bb62R883-R884

**Comment by koushikch7** [comment]
@ihor-sviziev its a simple code as below,
```$productCollectionFactory = $objectManager->get('\Magento\Catalog\Model\ResourceModel\Product\CollectionFactory');
$storeId = 10;
$productCollection = $productCollectionFactory->create()->addAttributeToSelect('*')->addStoreFilter($storeId);
foreach ($productCollection as $product) {
$sku = $product->getSku();
$product->setSku('UpdatedSku-'.$sku);
$product->save();
}

**Comment by ihor-sviziev** [comment]
@koushikch7 thank you! It would be much better to identify the issue. Just added this code sample to issue description

**Comment by ihor-sviziev** [comment]
Hi @koushikch7,
We have confirmation that PR fixes this issue: https://github.com/magento/magento2/pull/29349#issuecomment-684757994

**Comment by shyopat** [comment]
You should have to use product repository for editing the products
`\Magento\Catalog\Model\ProductRepository`

Use like below
```
$objectManager      = \Magento\Framework\App\ObjectManager::getInstance();
    $productCollection  = $objectManager->create('Magento\Catalog\Model\ResourceModel\Product\Collection');
    $productRepository  = $objectManager->get('\Magento\Catalog\Model\ProductRepository');
    $collection         = $productCollection->addAttributeToSelect('*')
                                            ->addFieldToFilter('type_id','simple')
                                            ->load();

    foreach ($collection as $product){
        
        $productRepository->save($product);     
    }
```

**Comment by koushikch7** [comment]
@shyopat  I tried the above snippet as well earlier but it was the same issue, but anyhow as per the ticket update I see the issue is fixed with the PR [#29349](https://github.com/magento/magento2/pull/29349#issuecomment-684757994) please check this from your end as well.
